specification/marketplace/resource-manager/Microsoft.Marketplace/stable/2021-12-01/Marketplace.json

```diff
@@ -0,0 +1,2894 @@
+            }
+          }
+        },
+        "x-ms-examples": {
+          "NotificationsState": {
+            "$ref": "./examples/NotificationsState.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/offers/{offerId}/acknowledgeNotification": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/PrivateStoreIdParameter"
+        },
+        {
+          "$ref": "#/parameters/OfferIdPathParameter"
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "PrivateStores"
+        ],
+        "operationId": "PrivateStore_AcknowledgeOfferNotification",
+        "description": "Acknowledge notification for offer",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/AcknowledgeOfferNotificationProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK, notification was acknowledged by the RP"
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "AcknowledgeNotification": {
+            "$ref": "./examples/AcknowledgeNotification.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/{privateStoreId}/requestApprovals/{requestApprovalId}/withdrawPlan": {
+      "parameters": [
+        {
+          "$ref": "#/parameters/PrivateStoreIdParameter"
+        },
+        {
+          "$ref": "#/parameters/RequestApprovalIdPathParameter"
+        },
+        {
+          "$ref": "#/parameters/ApiVersionParameter"
+        }
+      ],
+      "post": {
+        "tags": [
+          "PrivateStores"
+        ],
+        "operationId": "PrivateStore_WithdrawPlan",
+        "description": "Withdraw a user request approval on specific plan",
+        "parameters": [
+          {
+            "in": "body",
+            "name": "payload",
+            "schema": {
+              "$ref": "#/definitions/WithdrawProperties"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK"
+          },
+          "default": {
+            "description": "Microsoft.Marketplace error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "WithdrawPlan": {
+            "$ref": "./examples/WithdrawPlan.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Marketplace/privateStores/fetchAllSubscriptionsInTenant": {
```

Comment:
>fetchAllSubscriptionsInTenant [](http://example.com/codeflow?start=52&length=29)

This looks like a POST action, not GET. Did you intend to declare it as POST ? GETs are only applied to resource types.

Reply:
I sent an email to armapireview DL but did not get answer yet so I will ask my question here: 

The purpose of this API is to fetch ALL the subscriptions of the tenant ( unlike the get subscriptions API of subscriptions RP  ). 

On the one hand, Subscription is not our RP resource, so cannot (and don't need) expose put/delete
On the other hand, it is a pageable API and the call to the next page should always be GET.
(according to this [documentation](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-pageable)) 

So what is the correct way to design this API? 

Maybe something like:
GET providers/Microsoft.Marketplace/privetestores/{privateStoreId}/Subscriptions

Reply:
Yes, I think that's probably the best way of doing what you want. That creates a RO resource type: Microsoft.Marketplace/privateStores/subscriptions (note 'subscriptions' is lower/camel case. It also requires that the objects returned in your list follow ARM resource format with name, id, and type at the top level and everything else in a "properties" object.

You can also model this as a POST, but that requires you implement your own paging interfaces: Trying to use x-ms-pageable on a POST method still requires you to implement GET as you observed.

Regarding your question to armapireview DL: please submit questions to stackoverflow (you can also send the link to armapireview for visibility). Guidance for posting on SO is here: [https://armwiki.azurewebsites.net/troubleshooting/stackoverflow_guide.html#prerequisite](https://armwiki.azurewebsites.net/troubleshooting/stackoverflow_guide.html#prerequisite).

Reply:
As this is not our resource I will use the POST option. Committed the change

And thanks for the reference to SO :)